docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc

```diff
@@ -1,4 +1,322 @@
+      kind: Component
+----
+
+NOTE: You can repeat this pattern for every component in your application.  
+
+
+== Create an Overlay for your Application and Application-specific Component Configuration
+
+For each version or variant of your application as configured in the prior steps, you'll define an application overlay and any additional application-specific component patches.  
+
+We'll version this concrete application and set of patches in its own overlay folder in the application folder, in our case `overlay/application-a/v1-overlay` which will hold:
+
+. `application-patch.yaml`: our application overlay
+. `component-patch.yaml`: a version-specific patch for all components, typically a branch name
+. `exception-component-patch.yaml`: an example version-specific patch for a specific component or set of components
+. `kustomization.yaml`: a kustomization file that defines how the patches are applied to components
+
+Let's start with our `application-patch.yaml` at `overlay/application-a/v1-overlay/application-patch.yaml` (replacing values with your own):
+[source,yaml]
+----
+- op: replace
+  path: /metadata/name
+  value: application-a-v1
+- op: replace
+  path: /spec/description
+  value: "Pipeline for application-a v1"
+- op: replace
+  path: /spec/displayName
+  value: "application-a v1"
+----
+
+Followed by our override for components at `overlay/application-a/v1-overlay/component-patch.yaml`:
+[source,yaml]
+----
+- op: replace
+  path: /spec/application
+  value: application-a-v1 # Must match /metadata/name in application-patch.yaml
+- op: replace
+  path: /spec/source/git/revision
+  value: release-v1 # Replace with your target branch for all components
+----
+
+If you have any patches specific to this application revision that only impact a certain component or set of components, you can define another patch as defined in `overlay/application-a/v1-overlay/exception-component-patch-1.yaml`:
+[source,yaml]
+----
+- op: replace
+  path: /spec/source/git/revision
+  value: main # In this example, one of our components will build off of main, so we have to set it in a separate patch.
+----
+
+and finally we can set up our `overlay/application-a/v1-overlay/kustomization.yaml` to apply these patches correctly:
+[source,yaml]
+----
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+nameSuffix: v1 # Add a suffix to all resource names in the application for uniqueness
+resources:
+  - ../base
+
+patches:
+  - target:
+      kind: Application
+    path: application-patch.yaml
+  - target:
+      kind: Component
+    path: component-patch.yaml
+  - target:
+      kind: Component
+      name: component-b
+    path: exception-component-patch-1.yaml
+----
+
+NOTE: This kustomizaton applies a suffix to all resources, we recommend doing this to ensure uniqueness and make it easier to identify components and applications.  
+
+NOTE: You can define more than one exceptional patch and match component names through regex.  
```

Comment:
Are these NOTEs using the proper formats?

Reply:
Yes